zerver/models.py

```diff
@@ -247,10 +249,38 @@ def flush_realm_emoji(sender, **kwargs):
 post_save.connect(flush_realm_emoji, sender=RealmEmoji)
 post_delete.connect(flush_realm_emoji, sender=RealmEmoji)
 
+def filter_pattern_validator(value):
+    prefix = value[0]
+    pattern = value[1:]
+    regex = re.compile(r'(\(\?P<\w+>.+\))')
+
+    if not value:
+        raise ValidationError('Filter pattern cannot be empty')
+
+    if prefix not in settings.ALLOWED_FILTER_PREFIXES:
+        raise ValidationError(
+            'Filter pattern cannot start with `%s`, use one of the valid prefixes: %s' %
+            (value[0], ', '.join(settings.ALLOWED_FILTER_PREFIXES)))
+
+    try:
+        re.compile(pattern)
+    except:
+        # Regex is invalid
+        raise ValidationError('Invalid filter pattern')
+
+    if not regex.match(pattern):
+        raise ValidationError('Pattern format string must be in the following format: (?P<\w+>.+)')
```

Comment:
Might make sense to add a small test class for this validator to keep track of important examples we want to make sure work